### Домашняя работа: Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

In [1]:
import numpy as np

#### 1) Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [2]:
std = 16
mx = 80
n = 256
print(f'Ответ: {(mx - 2 * std / np.sqrt(n), mx + 2 * std / np.sqrt(n))}')

Ответ: (78.0, 82.0)


#### 2) В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

In [3]:
sample = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
mx = sample.mean()
print(f'Мат.ожидание: {mx}')
std = sample.std(ddof=1)
print(f'среднеквадратичное отклонение: {std}')

Мат.ожидание: 6.590000000000001
среднеквадратичное отклонение: 0.4508017549014448


по таблице критерия Стьюдента значение t равно 2.2622 (n-1 и доверительная вероятность 0.95)

In [5]:
t = 2.2622

построим доверительный интервал:

In [6]:
(mx - t * std / np.sqrt(n), mx + t * std / np.sqrt(n))

(6.526262266878873, 6.653737733121129)

данный интервал с вероятностью 0.95 определяет истинное значение измеряемой величины Х.

#### 3) Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

In [7]:
n = 100
mx = 17.5
scale = 2
# среднее значение имеет такое же распределение как случайная величина:
mu = 17
# исходя из центральной предельной теоремы получаем стандартное отклонение:
sigma = scale / np.sqrt(n)
# Заданный уровень значимости:
alpha = 0.025

строим интервал:

In [8]:
(mu + 2 * sigma, float('inf')) # правосторонная область

(17.4, inf)

Среднее значение выборки из 100 входит в правостороннюю критическую область 17.5 > 17.4, поэтому нулевая гипотеза, что шарики для подшипников имеют средний диметр 17 мм отвергается в пользу альтернативной гипотезы, что средний диаметр не равен 17 мм.

#### 4) Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [9]:
sample = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
mx = sample.mean()
std = sample.std(ddof=1)
n = 10

по таблице критерия Стьюдента значение t равно 3.2498 (n-1 и доверительная вероятность 0.99)

In [10]:
t = 3.2498

строим интервал:

In [11]:
(mx - t * std / np.sqrt(n), mx + t * std / np.sqrt(n))

(193.92327845730884, 203.07672154269116)

Утверждение продавца является верным, так как с вероятностью 99% средний вес пачки находится в интервале

от 193.92327845730884 до 203.07672154269116